libs/math/include/math/tensor.hpp

```diff
@@ -2200,6 +2204,23 @@ typename Tensor<T, C>::ConstSliceType Tensor<T, C>::Slice(std::vector<SizeType>
   return ConstSliceType(*this, range, axes);
 }
 
+template <typename T, typename C>
+typename Tensor<T, C>::ConstSliceType Tensor<T, C>::StridedSlice(
+    std::vector<SizeType> const &begins, std::vector<SizeType> const &ends,
+    std::vector<SizeType> const &strides) const
+{
+  std::vector<std::vector<SizeType>> range;
+  std::vector<SizeType>              axis;
+
+  for (SizeType j = 0; j < shape().size(); ++j)
+  {
+    range.push_back({begins.at(j), ends.at(j) + 1, strides.at(j)});
```

Comment:
shouldn't it be ends.at(j) without the + 1
And then you remove the -1 in the iterator?